### ISSUE TITLE
Settings -> Developer UI Tab - add ability to change git setting to a fork

### ISSUE THREAD

**Issue by OnlineDynamic** [FalconChristmas/fpp#1533]
It would be really helpful if it were possible to change fpp settings from the UI to point at a fork of the main fpp repository - and then select a branch on that fork to work from.  This would help those of us that like to test amd contribute code but aren't part of the core development team working in the main repo - who occasionaly raise pull requests to bring code across back to the main repo.

**Comment by Poporacer** [comment]
You mean something like this?
![image](https://github.com/FalconChristmas/fpp/assets/46263877/52bd9725-b95b-495a-b993-ae1cfea29eed)

**Comment by OnlineDynamic** [comment]
> You mean something like this? ![image](https://user-images.githubusercontent.com/46263877/254057761-52bd9725-b95b-495a-b993-ae1cfea29eed.png)

isn't that just the change log showing commits in the main fpp repo for the branch the user has selected from the UI config...?

I'm talking about easily pointing that fpp instance at a forked repo and branches in that forked repo

**Comment by dkulp** [comment]
IMO, this is something that would be super low priority.  If someone submitted a pull request, I wouldn't reject it, but not something I'm likely to spend much time on.   The main reason is that anyone that is likely creating forks and setting up development branches and pull requests and such it very likely to be comfortable with git on the command line would be more likely to be setting things up that way.

**Comment by Poporacer** [comment]
@OnlineDynamic No, that is link to jump to that revision, 
![image](https://github.com/FalconChristmas/fpp/assets/46263877/253c8d83-6ede-4e2f-afc2-433aa950159d)

**Comment by OnlineDynamic** [comment]
> IMO, this is something that would be super low priority. If someone submitted a pull request, I wouldn't reject it, but not something I'm likely to spend much time on. The main reason is that anyone that is likely creating forks and setting up development branches and pull requests and such it very likely to be comfortable with git on the command line would be more likely to be setting things up that way.

@dkulp I agree its a low priority nice to have... I was mainly getting frustrated by the script/git_pull constantly changing the origin back to standard with `        $SUDO git remote set-url origin "https://github.com/FalconChristmas/fpp.git"`

**Comment by cpinkham** [comment]
You could try something like the (untested) code below to check the existing origin url using "git remote get-url origin" and not change the setting if the origin IP/hostname is already set.

Here's the general idea:
```
    ORIGIN=$(git remote get-url origin | cut -f3 -d/)
    if [ "${ORIGIN}" != "${SOURCEIP}" ]; then
        if [ "${SOURCEIP}" == "github.com" ]; then
            $SUDO git remote set-url origin "https://github.com/FalconChristmas/fpp.git"
        else
            $SUDO git remote set-url origin "http://${SOURCEIP}/git/"
        fi
    fi
```

That would at least let you use forks on github without git_pull resetting the origin every time you pull.
